### DISCUSSION TITLE
Error when updating v5.4.0 > v5.5.0

### DISCUSSION THREAD

**Discussion by Colossus68** [LycheeOrg/Lychee#2489] [Bugs] [answered]
### Lychee version

v5.5.0

### Which PHP version are you using?

PHP 8.2

### Detailed description of the problem

I tried to update my v5.4 to v5.5 twice today. I received the following error message after logging in:

_500
HttpException
Target [Illuminate\Database\Migrations\MigrationRepositoryInterface] is not instantiable while building [App\Console\Commands\Laravel\MigrateFresh, Illuminate\Database\Migrations\Migrator]._

What is the problem and what can I do about it? 

For the time being I'll revert to v5.4.0

Thanks for your help

Peter

### Steps to reproduce the issue

1. Open URL
2. Login (Button "Migrate") > Error

### Diagnostics [REQUIRED]

Diagnostics
    -----------
        Warning: /var/www****************************************************************g/public/uploads/ has permissions 0755, but should have 2775
        Warning: /var/www****************************************************************g/public/uploads/small has permissions 0775, but should have 2775
        Warning: /var/www****************************************************************g/public/uploads/small/ef has permissions 0775, but should have 2775
        Warning: /var/www****************************************************************g/public/uploads/small/ef/e8 has permissions 0775, but should have 2775
        Warning: /var/www****************************************************************g/public/uploads/small/ef/34 has permissions 0775, but should have 2775
        Warning: 5966 more directories with wrong permissions
        Warning: Dropbox import not working. dropbox_key is empty.
        Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
        Warning: You may experience problems when uploading a photo of large size or handling many/large albums. Take a look in the FAQ for details.
        Error: Database is behind file version. Please apply the migrations.
        Info: Latest version of PHP is 8.3
    
    System Information
    ------------------
        Lychee Version (release):                5.5.0
        DB Version:                              5.4.0
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               false
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.2.7
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                Europe/Berlin
        Max uploaded file size:                  64M
        Max post size:                           64M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      120
        MySQL Version:                           8.0.26-16
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1690
        GD Version:                              bundled (2.1.0 compatible)
        Number of foreign key:                   12 found.
    
    Config Information
    ------------------
        version:                                 050400
        check_for_updates:                       0
        sorting_photos_col:                      taken_at
        sorting_photos_order:                    ASC
        sorting_albums_col:                      max_taken_at
        sorting_albums_order:                    DESC
        imagick:                                 1
        skip_duplicates:                         0
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        1920
        medium_max_height:                       1080
        lang:                                    de
        image_overlay_type:                      desc
        default_license:                         reserved
        compression_quality:                     90
        grants_full_photo_access:                0
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       10
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     1
        force_migration_in_production:           1
        search_public:                           0
        hide_version_number:                     0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         0
        photos_wraparound:                       1
        map_display:                             1
        zip64:                                   1
        map_display_public:                      1
        map_provider:                            OpenStreetMap.de
        force_32bit_ids:                         0
        map_include_subalbums:                   0
        update_check_every_days:                 3
        has_exiftool:                            0
        share_button_visible:                    0
        import_via_symlink:                      0
        has_ffmpeg:                              0
        location_decoding:                       1
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    1
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           0
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               0
        nsfw_warning:                            0
        nsfw_warning_admin:                      0
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        SA_enabled:                              1
        default_album_protection:                1
        album_decoration:                        all
        album_decoration_orientation:            column
        allow_username_change:                   1
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        layout:                                  masonry
        date_format_photo_thumb:                 d.m.Y, H:i:s
        date_format_photo_overlay:               d.m.Y, H:i:s
        date_format_sidebar_uploaded:            d.m.Y, H:i:s
        date_format_sidebar_taken_at:            d.m.Y, H:i:s
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             d.m.Y, H:i:s
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        default_album_thumb_aspect_ratio:        1/1
        use_album_compact_header:                0
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        current_job_processing_visible:          1
        public_recent:                           0
        public_starred:                          0
        SA_random_thumbs:                        0
        job_processing_queue_visible:            1
        login_required:                          0

### Browser & System [REQUIRED]

Client: Windows 11, Google  Chrome
Server: nginx, PHP8.2, MySQL 8.0

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment, marked as answer]
> I tried to update my v5.4 to v5.5 twice today. 

Can you describe your process here?

I think the problem is that you are overwriting the files. But in this specific case, there are some files that need to be removed (hence why it is breaking).
In summary, you can nuke the full app folder and replace it with folder from the release archive.

**Reply by Colossus68** [reply]
Yes, that's it! Now it works. So far I have only overwritten the old version with the new one.

Thank you very much!
